### PR TITLE
SafetyPolygon for the RSS visualization and ScenarioDumper extension

### PR DESCRIPTION
**Pull request by patrickhart**
* Extension of the ScenarioDumper. It is now possible to specify config_ids
* Drawing of SafetyPolygons that contain the lateral and longitudinal safety distances, as well as other additional information
* Python-bindings for the SafetyPolygon
* Modification of the RSSViewer to plot the SafetyPolygons
* Modification of the RSSEvaluator to store additional data
* New interfaces for the RSSEvaluator, SafetyPolygon, and BehaviorRSSConformant
* Created and extended the serialization of SafetyPolygon and BehaviorRSSConformant

### CODE REVIEW

**Review by juloberno — APPROVED** [review]
Looks good and can be merged. I especially like the idea to introduce the safety polygon as separate type.

File: bark/world/evaluation/rss/safety_polygon.hpp [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    return polygon;
+  }
+  AgentId GetAgentId() const {
+    return agent_id;
+  }
+};
+
+typedef std::shared_ptr<SafetyPolygon> SafetyPolygonPtr;
+
+inline void ComputeSafetyPolygon(
```
Comment (klemense1) [inline bark/world/evaluation/rss/safety_polygon.hpp:50]: what is this function doing? is it transforming a cartesian axis-aligned polygon to the orientation of the vehicle of the agent? can it be implemented using some base functions from bark::polygon (transform or so?)
either way, a doc-string might be beneficial.

**Review by klemense1 — CHANGES_REQUESTED** [review]
Docstring for ComputeSafetyPolygon() would be good.

File: bark/world/evaluation/rss/safety_polygon.hpp (hunk 1 above)
Comment (patrickhart) [reply bark/world/evaluation/rss/safety_polygon.hpp:50]: Okay, will add that. I checked and I did not find any appropriate geometry functions.
boost::geometry::scale_transformer e.g. transforms it in all dimensions.

**Review by klemense1 — APPROVED** [review]

**Pull request merged**